generator.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import argparse
+import subprocess
+import sys
+import asyncio
+
+DEFAULT_GIR_FILES_DIRECTORY = Path("./gir-files")
+DEFAULT_GIR_DIRECTORY = Path("./gir/")
+DEFAULT_GIR_PATH = DEFAULT_GIR_DIRECTORY / "target/release/gir"
+
+
+def run_command(command, folder=None):
+    return subprocess.run(command, cwd=folder, check=True)
+
+
+async def spawn_process(exe, args):
+    p = await asyncio.create_subprocess_exec(
+        str(exe),
+        *args,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+
+    stdout, stderr = await p.communicate()
+    assert p.returncode == 0, stderr.strip()
+    stdout = stdout.decode("utf-8")
+    stderr = stderr.decode("utf-8")
+    return stdout, stderr
+
+
+async def spawn_gir(gir_exe, args):
+    stdout, stderr = await spawn_process(gir_exe, args)
+    # Gir doesn't print anything to stdout. If it does, this is likely out of
+    # order with stderr, unless the printer/logging flushes in between.
+    assert not stdout, "`gir` printed unexpected stdout: {}".format(stdout)
+    return stderr
+
+
+def update_workspace():
+    return run_command(["cargo", "build", "--release"], "gir")
+
+
+def ask_yes_no_question(question, conf):
+    question = "{} [y/N] ".format(question)
+    if conf.yes:
+        print(question + "y")
+        return True
+    line = input(question)
+    return line.strip().lower() == "y"
+
+
+def update_submodule(submodule_path, conf):
+    if any(submodule_path.iterdir()):
+        return False
+    print("=> Initializing {} submodule...".format(submodule_path))
+    run_command(["git", "submodule", "update", "--init", submodule_path])
+    print("<= Done!")
+
+    if ask_yes_no_question(
+        "Do you want to update {} submodule?".format(submodule_path), conf
+    ):
+        print("=> Updating submodule...")
+        run_command(["git", "reset", "--hard", "HEAD"], submodule_path)
+        run_command(["git", "pull", "-f", "origin", "master"], submodule_path)
+        print("<= Done!")
+        return True
+    return False
+
+
+def build_gir():
+    print("=> Building gir...")
+    update_workspace()
+    print("<= Done!")
+
+
+async def regenerate_crate_docs(gir_exe, crate_dir, base_gir_args):
```

Comment:
Since we are now going to generate docs only from our side and not use lgpl-docs anymore, should we make `gir` generates doc comments directly into the code?

cc @sdroege @bilelmoussaoui

Reply:
As we run `gir` anyway that seems like a good idea... but also will require quite some work on gir. Let's do that as a second step after merging to get rid of rustdoc-stripper completely?

Reply:
Not sure it would require that much work. But can be done as a second step.

Reply:
Good, I like incremental improvements :)